src/gleam/list.gleam

```diff
@@ -1865,7 +1865,7 @@ pub fn permutations(l: List(a)) -> List(List(a)) {
 fn do_window(acc: List(List(a)), l: List(a), n: Int) -> List(List(a)) {
   let window = take(l, n)
 
-  case length(window) == n {
+  case n != 0 && length(window) == n {
```

Comment:
Since n doesn't change throughout the recursion we could move this check up into `window` itself instead of repeating it here:
```gleam
pub fn window(list, n) {
  case n <= 0 { ... }
}
```